### PR TITLE
Always check for error when (un)mounting

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Both `mnt_context_mount` and `mnt_context_umount` can return zero
even if the (un)mount failed. This happens when using the (un)mount
helper program (e.g. mount.ntfs), in this case we need to get the
return code using `mnt_context_get_helper_status`.

### CODE REVIEW

File: src/plugins/fs/mount.c [hunk 1]
```diff
@@ -72,6 +73,14 @@ static void parse_unmount_error_old (struct libmnt_context *cxt, int rc, const g
...
                 break;
         }
+    } else if (mnt_context_helper_executed (cxt) == 1) {
+        helper_status = mnt_context_get_helper_status (cxt);
+        if (helper_status != 0) {
+            g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_FAIL,
+                         "Unmount helper program failed: %d.", helper_status);
+            return FALSE;
+        } else
+            return TRUE;
```
Comment (vpodzime) [inline src/plugins/fs/mount.c]: Are you sure this is the only place where this function should return ``TRUE``?

File: src/plugins/fs/mount.c [hunk 2]
```diff
@@ -259,6 +263,14 @@ static gboolean parse_mount_error_old (struct libmnt_context *cxt, int rc, Mount
...
                 break;
         }
+    } else if (mnt_context_helper_executed (cxt) == 1) {
+        helper_status = mnt_context_get_helper_status (cxt);
+        if (helper_status != 0) {
+            g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_FAIL,
+                         "Mount helper program failed: %d.", helper_status);
+            return FALSE;
+        } else
+            return TRUE;
```
Comment (vpodzime) [inline src/plugins/fs/mount.c]: Same here.

File: tests/fs_test.py [hunk 3]
```diff
@@ -86,6 +86,16 @@ def _clean_up(self):
...
 
+    def setro(self, device):
+        ret, _out, _err = utils.run_command("blockdev --setro %s" % device)
+        if ret != 0:
+            self.fail("Failed to set %s read-only" % device)
+
+    def setrw(self, device):
+        ret, _out, _err = utils.run_command("blockdev --setrw %s" % device)
+        if ret != 0:
+            self.fail("Failed to set %s read-only" % device)
```
Comment (vpodzime) [inline tests/fs_test.py]: *read-write*

File: tests/fs_test.py [hunk 4]
```diff
@@ -1125,6 +1135,67 @@ def test_mount_fstab_user(self):
...
+
+        # set the device read-only
+        self.setro(self.loop_dev)
+        self.addCleanup(self.setrw, self.loop_dev)
+
+        # standard mount (rw) should fail
+        with self.assertRaises(GLib.GError):
+            BlockDev.fs_mount(self.loop_dev, tmp, "ntfs", None)
+
+        # read-only mount shoudl work
```
Comment (vpodzime) [inline tests/fs_test.py]: *should*

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Pull request merged**